File: solutions/3335. Total Characters in String After Transformations I/3335.py
class Solution:
  def lengthAfterTransformations(self, s: str, t: int) -> int:
    MOD = 1_000_000_007
    count = [0] * 26

    for c in s:
      count[ord(c) - ord('a')] += 1

    for _ in range(t):
      newCount = [0] * 26
      # 'a' -> 'b', 'b' -> 'c', ..., 'y' -> 'z'
      for i in range(25):
        newCount[i + 1] = count[i]
      # 'z' -> 'ab'
      newCount[0] = count[25]
      newCount[1] = (newCount[1] + count[25]) % MOD
      count = newCount

    return sum(count) % MOD

File: solutions/3335. Total Characters in String After Transformations I/3335.java
class Solution {
  public int lengthAfterTransformations(String s, int t) {
    final int MOD = 1_000_000_007;
    int[] count = new int[26];

    for (final char c : s.toCharArray())
      ++count[c - 'a'];

    while (t-- > 0) {
      int[] newCount = new int[26];
      // 'a' -> 'b', 'b' -> 'c', ..., 'y' -> 'z'
      for (int i = 0; i < 25; i++)
        newCount[i + 1] = count[i];
      // 'z' -> 'ab'
      newCount[0] = count[25];
      newCount[1] = (newCount[1] + count[25]) % MOD;
      count = newCount;
    }

    return Arrays.stream(count).reduce(0, (a, b) -> (a + b) % MOD);
  }
}
